### PR TITLE
[starts #187584911] Seller statistics per timeframe

### PR DESCRIPTION
**Pull request by ProgrammerDATCH**

### CODE REVIEW

**Review by MANISHIMWESalton — DISMISSED** [review]

**Review by ndahimana154 — DISMISSED** [review]

File: src/routes/productRouter.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Router } from "express";
+import productControllers from "../modules/product/controller/productControllers";
+import { validation } from "../middlewares/validation";
+import { statisticsSchema } from "../modules/product/validation/productValidations";
+import { userAuthorization } from "../middlewares/authorization";
+
+const router: Router = Router()
+
+router.post("/statistics", validation(statisticsSchema), userAuthorization(["buyer", "seller", "admin"]), productControllers.getSellerStatistics)
```
Comment (key-joshua) [inline src/routes/productRouter.ts]: This middleware should take only the seller role.

File: src/databases/models/cart.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+class Cart extends Model<CartAttributes> implements CartAttributes {
+    declare id: number;
+    declare userId: number;
+    declare status: string;
+    declare createdAt: Date;
+    declare updatedAt: Date;
+
+    static associate(models: any) {
+        Cart.belongsTo(models.User, { foreignKey: "userId", as: "user" });
```
Comment (key-joshua) [inline src/databases/models/cart.ts]: Let's **`as: buyer`** instead

**Comment by key-joshua** [comment]
We don't have this relationship within users tables. **`Users.hasMany(models.Products, { foreignKey: "sellerId", as: "products" });`**

File: src/databases/models/products.ts [hunk 3]
```diff
@@ -1,111 +1,111 @@
...
-    declare expired: boolean;
-    declare bonus?: string;
-    declare images: string[];
-    declare quantity: number;
-    declare isAvailable: string;
-    declare createdAt: Date;
-    declare updatedAt: Date;
-
-    static associate(models: any) {
-        Products.belongsTo(models.Users, { foreignKey: "userId", as: "user" });
```
Comment (key-joshua) [inline src/databases/models/products.ts:25]: We don't have this relationship  **`Products.belongsTo(models.Users, { foreignKey: "userId", as: "user" });`**

**Pull request merged**